### DISCUSSION TITLE
Runtime sitemap in resource route

### DISCUSSION THREAD

**Discussion by gerbyzation** [fedeya/remix-sitemap#37] [General]
Looking at the example it recommends to setup the sitemap in the `handleRequest` in `entry.server.ts`. Curious why this is handled in `entry.server.ts` instead of a resource route? Not having it part of the file based router makes this less transparent and harder to find. Appreciate there might be constraints I'm not aware of, so curious to understand the reasoning behind this better. If there are any technical limitations or considerations pushing it to `entry.server.ts` would be great if that can be documented in the README for context

**Comment by fedeya** [comment]
Hi @gerbyzation!

This library makes use of the `remixContext` to get the project routes data (the path, `export sitemap` function and etc) to generate the sitemap and it's context is not exported to loaders functions
